cmd/thermal-recorder/main.go

```diff
@@ -225,10 +232,10 @@ func handleConn(conn net.Conn, conf *Config, turret *TurretController, recording
 			if err != nil {
 				return err
 			}
-			numFrames++
+			framesWritten++
 		}
 
-		frame, prevFrame = prevFrame, frame
+		frame, prevFrame = frameLoop.MoveToNextFrame(), frame
```

Comment:
instead of tracking `prevFrame` in this function could there be a `frameLoop.Previous()` which returns the previous frame?